src/data-pipeline/spark-etl/src/main/java/software/aws/solution/clickstream/IPEnrichment.java

```diff
@@ -57,7 +58,7 @@ public Dataset<Row> transform(final Dataset<Row> dataset) {
         ));
         Dataset<Row> ipEnrichDataset = dataset.withColumn("geo",
                 udfEnrichIP.apply(
-                        col(GEO_FOR_ENRICH).getItem("ip"),
+                        split(col(GEO_FOR_ENRICH).getItem("ip"), ",").getItem(0),
```

Comment:
Is it safe to process the empty `ip` field? Let's have a test for it if it's not covered.

Reply:
It is safe for empty ip. Added test case for empty ip.